code/ATMOSPHERICS/components/unary_devices/vent_pump.dm

```diff
@@ -309,13 +309,6 @@
 		stat |= NOPOWER
 	update_icon_nopipes()
 
-/obj/machinery/atmospherics/components/unary/vent_pump/Destroy()
-	if(initial_loc)
-		initial_loc.air_vent_info -= id_tag
-		initial_loc.air_vent_names -= id_tag
-	..()
```

Comment:
There's another Destroy() defined further up in the file for the same path that already does all this. This one was redundant and messed up the return value.